distributed/scheduler.py

```diff
@@ -4374,6 +4374,9 @@ async def add_worker(
                     await comm.write(msg)
                 return
 
+            self.log_event(address, {"action": "add-worker"})
+            self.log_event("all", {"action": "add-worker", "worker": address})
```

Comment:
Moving these to the beginning of the transaction helps debugging causation a bit